css/design_system/utils/colors.scss

```
@@ -0,0 +1,55 @@
+$colors: 
+  "--s-color-black", 
+  "--s-color-white", 
+  "--s-color-cyan-400",
+  "--s-color-cyan-500",
+  "--s-color-cyan-600",
+  "--s-color-cyan-700",
+  "--s-color-gray-100",
+  "--s-color-gray-200",
+  "--s-color-gray-300",
+  "--s-color-gray-400",
+  "--s-color-gray-500",
+  "--s-color-gray-600",
+  "--s-color-gray-700",
+  "--s-color-gray-800",
+  "--s-color-red-500",
+  "--s-color-red-600",
+  "--s-color-red-700",
+  "--s-color-yellow-500",
+  "--s-color-yellow-600",
+  "--s-color-yellow-700";
+
+@mixin docstring($class, $util_type, $color_type, $text) {
+  /**
+   * @title #{$class}
+   * @category utils
+   * @util_type #{$util_type}
+   * @color_group #{$color_type}
+   *
+   * @example
+   * <div class="#{$class}">#{$text}</div>
+   */
+}
```

I think you could get away with _no_ updates to the `css/doc_site_styling.scss` if you apply other classes to the example div here instead. Here's an idea for an update to this mixin:

```suggestion
@mixin docstring($class, $util_type, $color_type, $text, $helper_classes) {
  /**
   * @title #{$class}
   * @category utils
   * @util_type #{$util_type}
   * @color_group #{$color_type}
   *
   * @example
   * <div class="#{$class} #{$helper_classes}">#{$text}</div>
   */
}
```

And then, for your text utils that need a dark background you could use:

```scss
@include docstring($text-class-name, "color", "text", "Example Text", "s-p-3 s-bg-black");
```

And for the border/background color you could include text (to ensure height) and apply the border width needed:

```scss
@include docstring($border-class-name, "color", "border", "Example Text", "s-p-3 s-border-width-2");
```

---

Ha ok! I went down this route for take 1. I only moved away from it because it made the example HTML have that extra class, which felt like clutter on the doc page itself, and I wasn't sure if that'd be a pain (if user is copy and pasting the code snippet, do they then need to delete that class or do they leave it and it bloats their code? or - would they likely not be copy and pasting - I don't know!).  With that, what do you think/prefer??

---

It's a good question, and I don't know if there's one right answer. My reasoning is just that (as a user of Savile) I'd prefer transparency over ease-of-copy-pasting. So when I look at the example code and see the rendered version, What I See Is What I Get™ and there isn't any "hidden" styling going on.

If that means I have to edit a few things after the fact, I'm ok with that. Since the utility classes are so well named, it should be super-obvious which classes are not necessary.

---

Ok I'm into that. Thanks!!

---

@tannerwelsh let's see what you think of this with the `if`. With the path I was on, I had to add the argument of `$example-class` into the mixin, because without it, the class name in the "class" column was inaccurate.